functional/sev/run.sh

```diff
@@ -0,0 +1,337 @@
+#!/bin/bash
+# Copyright 2022 Advanced Micro Devices, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -eE
+set -o nounset
+set -o pipefail
+
+trap cleanup EXIT
+
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+NC='\033[0m'
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+tests_repo_dir="$(realpath "$script_dir/../..")"
+
+export EDK2_REPO_DIR="${HOME}/edk2"
+export DOCKER_USER="rksavino"
+
+export KBS_DB_HOST="127.0.0.1"
+export KBS_DB_USER="root"
+export KBS_DB_PW="root"
+export KBS_DB="simple_kbs"
+export KBS_DB_TYPE="mysql"
+
+esudo() {
+  sudo -E PATH=$PATH "$@"
+}
+
+modify_kata_config() {
+  esudo mkdir -p /etc/kata-containers/
+  esudo install -o root -g root -m 0640 /usr/share/defaults/kata-containers/configuration.toml /etc/kata-containers
+
+  # Enable full debug
+  esudo sed -i -e 's/^# *\(enable_debug\).*=.*$/\1 = true/g' /etc/kata-containers/configuration.toml
+  
+  # kernel, initrd, image, firmware
+  esudo sed -i -e 's/kernel = "\(.*\)"/kernel = "\/usr\/share\/kata-containers\/vmlinuz-sev.container"/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e 's/initrd = "\(.*\)"/initrd = "\/usr\/share\/kata-containers\/kata-containers-initrd.img"/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e 's/image = "\(.*\)"/#image = ""\ninitrd = "\/usr\/share\/kata-containers\/kata-containers-initrd.img"/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e "s|^\(firmware = \"\).*\"|\1${EDK2_REPO_DIR}/Build/AmdSev/DEBUG_GCC5/FV/OVMF.fd\"|g" /etc/kata-containers/configuration.toml
+  
+  # kernel_params
+  esudo sed -i -e 's/kernel_params = "\(.*\)"/kernel_params = " agent.log=debug initcall_debug agent.config_file=\/etc\/default\/agent.cfg console=ttyAMA0,115200 console=tty highres=off console=ttyS0"/g' /etc/kata-containers/configuration.toml
+
+  # confidential_guest, guest_attestation
+  esudo sed -i -e 's/# confidential_guest = true/confidential_guest = true/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e 's/#guest_pre_attestation = true/guest_pre_attestation = true/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e 's/#guest_pre_attestation_proxy="localhost:50051"/guest_pre_attestation_proxy = "localhost:44444"/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e 's/#guest_pre_attestation_keyset="KEYSET-1"/guest_pre_attestation_keyset = "KEYSET-1"/g' /etc/kata-containers/configuration.toml
```

Comment:
At some point we should test out setting this via annotation rather than in the config file.

Reply:
Annotation meaning the `Makefile` variables for the sev config template?

Reply:
I think these options can be set via a Kubernetes annotation, which for some fields (like `keyset`) is how a user will probably set them. Keyset and proxy should be changed at the scope of the pod rather than the scope of the node. So at some point we might want to adjust the testing to account for this.

Reply:
This part should be done via the configuration file, for values that we know to be the default ones, in the qemu-sev.conf

Reply:
@fitzthum all the guest_pre_attestation_* and sev_cert_chain can be passed via pod annotation?